inc/taskjobstate.class.php

```diff
@@ -419,35 +419,41 @@ public function ajaxGetLogs($params)
     public function getLogs($id, $last_date)
     {
         global $DB;
-        $fields = [
-        'log.id'      => 0,
-        'log.date'    => 1,
-        'log.comment' => 2,
-        'log.state'   => 3,
-        'run.id'      => 4,
-        ];
-        $query = "SELECT log.`id` AS 'log.id',
-                  log.`date` AS 'log.date',
-                  log.`comment` AS 'log.comment',
-                  log.`state` AS 'log.state',
-                  run.`uniqid` AS 'run.id'
-                FROM `glpi_plugin_glpiinventory_taskjoblogs` AS log
-                LEFT JOIN `glpi_plugin_glpiinventory_taskjobstates` AS run
-                  ON run.`id` = log.`plugin_glpiinventory_taskjobstates_id`
-                WHERE run.`id` = $id
-                  AND log.`date` <= '$last_date'
-               ORDER BY log.`id` DESC";
-        $res = $DB->query($query);
+
+        $iterator = $DB->request([
+            'SELECT' => [
+                'log.id',
+                'log.date',
+                'log.comment',
+                'log.state',
+                'run.uniqid AS runid'
+            ],
+            'FROM'   => 'glpi_plugin_glpiinventory_taskjoblogs AS log',
+            'LEFT JOIN' => [
+                'glpi_plugin_glpiinventory_taskjobstates AS run' => [
+                    'ON' => [
+                        'run' => 'id',
+                        'log' => 'plugin_glpiinventory_taskjobstates_id'
+                    ]
+                ]
+            ],
+            'WHERE'  => [
+                'run.id' => $id,
+                'log.date' => ['<=' => $last_date]
+            ],
+            'ORDER'  => 'log.id DESC'
+        ]);
+
```

Comment:
Get some warning from ```deploy``` task (From computer tab)

![image](https://user-images.githubusercontent.com/7335054/236144966-3fabc165-30f1-4867-a063-74ccea0cf9f8.png)

Logs : 

```
[2023-05-04 09:58:29] glpiphplog.WARNING:   *** PHP Warning (2): Undefined array key "logs" in /home/TECLIB/DEV/GLPI/10.0-bugfixes/plugins/glpiinventory/inc/deploypackage.class.php at line 2058
  Backtrace :
  .../glpiinventory/inc/deploypackage.class.php:1813 PluginGlpiinventoryDeployPackage->getMyDepoyPackagesState()
  .../glpiinventory/inc/deploypackage.class.php:1414 PluginGlpiinventoryDeployPackage->getPackageForMe()
  .../glpiinventory/inc/deploypackage.class.php:1097 PluginGlpiinventoryDeployPackage->showPackageForMe()
  src/CommonGLPI.php:689                             PluginGlpiinventoryDeployPackage::displayTabContentForItem()
  ajax/common.tabs.php:116                           CommonGLPI::displayStandardTab()
  public/index.php:82                                require()
  
[2023-05-04 09:58:29] glpiphplog.WARNING:   *** PHP Warning (2): Trying to access array offset on value of type null in /home/TECLIB/DEV/GLPI/10.0-bugfixes/plugins/glpiinventory/inc/deploypackage.class.php at line 2058
  Backtrace :
  .../glpiinventory/inc/deploypackage.class.php:1813 PluginGlpiinventoryDeployPackage->getMyDepoyPackagesState()
  .../glpiinventory/inc/deploypackage.class.php:1414 PluginGlpiinventoryDeployPackage->getPackageForMe()
  .../glpiinventory/inc/deploypackage.class.php:1097 PluginGlpiinventoryDeployPackage->showPackageForMe()
  src/CommonGLPI.php:689                             PluginGlpiinventoryDeployPackage::displayTabContentForItem()
  ajax/common.tabs.php:116                           CommonGLPI::displayStandardTab()
  public/index.php:82                                require()
  
[2023-05-04 09:58:29] glpiphplog.WARNING:   *** PHP Warning (2): Trying to access array offset on value of type null in /home/TECLIB/DEV/GLPI/10.0-bugfixes/plugins/glpiinventory/inc/deploypackage.class.php at line 2058
  Backtrace :
  .../glpiinventory/inc/deploypackage.class.php:1813 PluginGlpiinventoryDeployPackage->getMyDepoyPackagesState()
  .../glpiinventory/inc/deploypackage.class.php:1414 PluginGlpiinventoryDeployPackage->getPackageForMe()
  .../glpiinventory/inc/deploypackage.class.php:1097 PluginGlpiinventoryDeployPackage->showPackageForMe()
  src/CommonGLPI.php:689                             PluginGlpiinventoryDeployPackage::displayTabContentForItem()
  ajax/common.tabs.php:116                           CommonGLPI::displayStandardTab()
  public/index.php:82                                require()
  
[2023-05-04 09:58:29] glpiphplog.WARNING:   *** PHP Warning (2): Undefined array key "logs" in /home/TECLIB/DEV/GLPI/10.0-bugfixes/plugins/glpiinventory/inc/deploypackage.class.php at line 2059
  Backtrace :
  .../glpiinventory/inc/deploypackage.class.php:1813 PluginGlpiinventoryDeployPackage->getMyDepoyPackagesState()
  .../glpiinventory/inc/deploypackage.class.php:1414 PluginGlpiinventoryDeployPackage->getPackageForMe()
  .../glpiinventory/inc/deploypackage.class.php:1097 PluginGlpiinventoryDeployPackage->showPackageForMe()
  src/CommonGLPI.php:689                             PluginGlpiinventoryDeployPackage::displayTabContentForItem()
  ajax/common.tabs.php:116                           CommonGLPI::displayStandardTab()
  public/index.php:82                                require()
  
[2023-05-04 09:58:29] glpiphplog.WARNING:   *** PHP Warning (2): foreach() argument must be of type array|object, null given in /home/TECLIB/DEV/GLPI/10.0-bugfixes/plugins/glpiinventory/inc/deploypackage.class.php at line 1521
  Backtrace :
  .../glpiinventory/inc/deploypackage.class.php:1097 PluginGlpiinventoryDeployPackage->showPackageForMe()
  src/CommonGLPI.php:689                             PluginGlpiinventoryDeployPackage::displayTabContentForItem()
  ajax/common.tabs.php:116                           CommonGLPI::displayStandardTab()
  public/index.php:82                                require()
  
[2023-05-04 09:59:38] glpiphplog.WARNING:   *** PHP Warning (2): Undefined array key "logs" in /home/TECLIB/DEV/GLPI/10.0-bugfixes/plugins/glpiinventory/inc/deploypackage.class.php at line 2058
  Backtrace :
  .../glpiinventory/inc/deploypackage.class.php:1813 PluginGlpiinventoryDeployPackage->getMyDepoyPackagesState()
  .../glpiinventory/inc/deploypackage.class.php:1414 PluginGlpiinventoryDeployPackage->getPackageForMe()
  .../glpiinventory/inc/deploypackage.class.php:1097 PluginGlpiinventoryDeployPackage->showPackageForMe()
  src/CommonGLPI.php:689                             PluginGlpiinventoryDeployPackage::displayTabContentForItem()
  ajax/common.tabs.php:116                           CommonGLPI::displayStandardTab()
  public/index.php:82                                require()
  
[2023-05-04 09:59:38] glpiphplog.WARNING:   *** PHP Warning (2): Trying to access array offset on value of type null in /home/TECLIB/DEV/GLPI/10.0-bugfixes/plugins/glpiinventory/inc/deploypackage.class.php at line 2058
  Backtrace :
  .../glpiinventory/inc/deploypackage.class.php:1813 PluginGlpiinventoryDeployPackage->getMyDepoyPackagesState()
  .../glpiinventory/inc/deploypackage.class.php:1414 PluginGlpiinventoryDeployPackage->getPackageForMe()
  .../glpiinventory/inc/deploypackage.class.php:1097 PluginGlpiinventoryDeployPackage->showPackageForMe()
  src/CommonGLPI.php:689                             PluginGlpiinventoryDeployPackage::displayTabContentForItem()
  ajax/common.tabs.php:116                           CommonGLPI::displayStandardTab()
  public/index.php:82                                require()
  
[2023-05-04 09:59:38] glpiphplog.WARNING:   *** PHP Warning (2): Trying to access array offset on value of type null in /home/TECLIB/DEV/GLPI/10.0-bugfixes/plugins/glpiinventory/inc/deploypackage.class.php at line 2058
  Backtrace :
  .../glpiinventory/inc/deploypackage.class.php:1813 PluginGlpiinventoryDeployPackage->getMyDepoyPackagesState()
  .../glpiinventory/inc/deploypackage.class.php:1414 PluginGlpiinventoryDeployPackage->getPackageForMe()
  .../glpiinventory/inc/deploypackage.class.php:1097 PluginGlpiinventoryDeployPackage->showPackageForMe()
  src/CommonGLPI.php:689                             PluginGlpiinventoryDeployPackage::displayTabContentForItem()
  ajax/common.tabs.php:116                           CommonGLPI::displayStandardTab()
  public/index.php:82                                require()
  
[2023-05-04 09:59:38] glpiphplog.WARNING:   *** PHP Warning (2): Undefined array key "logs" in /home/TECLIB/DEV/GLPI/10.0-bugfixes/plugins/glpiinventory/inc/deploypackage.class.php at line 2059
  Backtrace :
  .../glpiinventory/inc/deploypackage.class.php:1813 PluginGlpiinventoryDeployPackage->getMyDepoyPackagesState()
  .../glpiinventory/inc/deploypackage.class.php:1414 PluginGlpiinventoryDeployPackage->getPackageForMe()
  .../glpiinventory/inc/deploypackage.class.php:1097 PluginGlpiinventoryDeployPackage->showPackageForMe()
  src/CommonGLPI.php:689                             PluginGlpiinventoryDeployPackage::displayTabContentForItem()
  ajax/common.tabs.php:116                           CommonGLPI::displayStandardTab()
  public/index.php:82                                require()
  
[2023-05-04 09:59:38] glpiphplog.WARNING:   *** PHP Warning (2): foreach() argument must be of type array|object, null given in /home/TECLIB/DEV/GLPI/10.0-bugfixes/plugins/glpiinventory/inc/deploypackage.class.php at line 1521
  Backtrace :
  .../glpiinventory/inc/deploypackage.class.php:1097 PluginGlpiinventoryDeployPackage->showPackageForMe()
  src/CommonGLPI.php:689                             PluginGlpiinventoryDeployPackage::displayTabContentForItem()
  ajax/common.tabs.php:116                           CommonGLPI::displayStandardTab()
  public/index.php:82                                require()
  
```

Reply:
Guess this is fixed as of c953f3d18924109f503e7128c1a9b4ae2dce1c78